lib/Dialect/TTNN/Transforms/Passes.cpp

```diff
@@ -181,8 +181,11 @@ class ConvertTTIRToTTNN
 void createTTIRToTTNNBackendPipeline(
     OpPassManager &pm, const TTIRToTTNNBackendPipelineOptions &options) {
   pm.addPass(mlir::tt::ttir::createTTIRLayout());
+
   if (options.gridSetPassEnabled) {
-    pm.addPass(mlir::tt::ttir::createTTIRGridSet());
+    ttir::TTIRGridSetOptions gridSetOptions;
+    gridSetOptions.overrideGridSizes = options.overrideGridSizes;
+    pm.addPass(mlir::tt::ttir::createTTIRGridSet(gridSetOptions));
   }
```

Comment:
Not related to this review, but leaving my thoughts here:
Is it possible to make the options argument optional in createTTIRToTTNNBackendPipeline? For example, I am envisioning a scenario where we prioritize making defaults work and calling it from TT-Forge like this:
createTTIRToTTNNBackendPipeline(pm); // Invoke TTIRToTTNNBackendPipelineOptions object with defaults

When we encounter integration issues, we can then override some options to allow things through. However, generally, I don't see the need to make this mandatory. Once we detect issues in integration, we can specific override some options to let things through, but in general I don't see a need to make this

Reply:
I think this is somehow already handled by framework because you can run from CL without specifying options, its an optional argument. Also all options have default inits defined:
![image](https://github.com/user-attachments/assets/de9c84ba-eb97-455e-8ba4-6d113f84c128)

Reply:
Digging a bit I am not finding need for default constructor invocation for pipeline/pass options. https://mlir.llvm.org/docs/PassManagement/#instance-specific-pass-options
Will investigate further but we should probably sync on what kind of problem exactly is this creating for you.